src/main/java/bio/overture/ego/service/DefaultProviderService.java

```diff
@@ -0,0 +1,31 @@
+package bio.overture.ego.service;
+
+import static bio.overture.ego.model.exceptions.NotFoundException.buildNotFoundException;
+
+import bio.overture.ego.model.entity.DefaultProvider;
+import bio.overture.ego.model.enums.ProviderType;
+import bio.overture.ego.repository.DefaultProviderRepository;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class DefaultProviderService extends AbstractBaseService<DefaultProvider, ProviderType> {
+
+  private final DefaultProviderRepository defaultProviderRepository;
+
+  @Autowired
+  public DefaultProviderService(@NonNull DefaultProviderRepository defaultProviderRepository) {
+    super(DefaultProvider.class, defaultProviderRepository);
+    this.defaultProviderRepository = defaultProviderRepository;
+  }
+
+  @Override
+  public DefaultProvider getWithRelationships(ProviderType id) {
```

Comment:
add `@NonNull` to the parameter. It is good practice for public method parameters to have the `@NonNull` annottion, since the content of the incomming parameter is out of context of this class. For instance, the developer, cannot control how the public method is used outside this class. Private methods on the otherhand, are in the developer's control, and so the use of `@NonNull` is optional, if the scenario makes sense.

The point is, we want to catch null errors as early as possible in the callstack, so that debugging is easier and the fix can be done faster.